packages/api/src/external/carequality/document/document-query-retrieval.ts

```diff
@@ -0,0 +1,65 @@
+import { PurposeOfUse } from "@metriport/shared";
+import { DocumentRetrievalReqToExternalGW, DocumentReference } from "@metriport/ihe-gateway-sdk";
+import dayjs from "dayjs";
+import { DocumentWithMetriportId } from "./shared";
+import { DocumentQueryResult } from "../document-query-result";
+import { Organization } from "@metriport/core/domain/organization";
+
+const SUBJECT_ROLE_CODE = "106331006";
+const SUBJECT_ROLE_DISPLAY = "Administrative AND/OR managerial worker";
+
+export function createCQDocumentRetrievalRequests({
+  requestId,
+  cxId,
+  organization,
+  documentReferences,
+  documentQueryResults,
+}: {
+  requestId: string;
+  cxId: string;
+  organization: Organization;
+  documentReferences: DocumentWithMetriportId[];
+  documentQueryResults: DocumentQueryResult[];
+}): DocumentRetrievalReqToExternalGW[] {
+  const orgOid = organization.oid;
+  const orgName = organization.data.name;
+  const user = `${orgName} System User`;
+  const now = dayjs().toISOString();
+
+  const requests: DocumentRetrievalReqToExternalGW[] = documentQueryResults.map(
+    documentQueryResult => {
+      const { patientId, gateway } = documentQueryResult.data;
+
+      const requestDocReferences: DocumentReference[] = documentReferences.filter(
+        docRef => docRef.homeCommunityId === gateway?.homeCommunityId
+      );
+
+      return {
+        id: requestId,
+        cxId: cxId,
+        patientId: patientId,
+        timestamp: now,
+        samlAttributes: {
+          subjectId: user,
+          subjectRole: {
+            code: SUBJECT_ROLE_CODE,
+            display: SUBJECT_ROLE_DISPLAY,
+          },
```

Comment:
we're not sending the NPI in the request to IHE GW? expected that to be something we send in all interfaces

as a side note, what's the rationale behind using `106331006` for the role code?

Reply:
It was not something that was added to the interfaces on notion so i assumed we didnt need them. As for the snomed code i just chose it as a generic role to always be passed https://www.findacode.com/snomed/106331006--administrative-andor-managerial-worker.html